docs/deployment-integration.md

```diff
@@ -0,0 +1,312 @@
+  accept finality signatures and randomness
+
+<!-- TODO: once we have docs for the allow list and rate limiting, we can link
+those here -->
+
+## Rollup BSN Registration
+<img width="3032" height="287" alt="register" src="https://github.com/user-attachments/assets/935fb9bd-f42d-47ba-a0f3-0fc628961e67" />
+
+After deploying and instantiating the Rollup BSN contract, the
+rollup must register on Babylon Genesis. Registration requires 
+submitting metadata to identify and describe the BSN
+
+To register the BSN, use:
+```shell
+babylond tx btcstkconsumer register-consumer \
+                                <consumer_id> \
+                                <consumer_name> \
+                                <consumer_ description> \
+                                <max_multi_staked_fps> \
+                                <rollup_finality_contract_address>
+```
+
+Required metadata for BSN registration:
+* `consumer_id`: BSN ID of your rollup (same as the one you used on contract
+  instantiation)
+* `consumer_name`: Human-readable name of your rollup BSN (e.g., `"DeFi Rollup Chain"`)
+* `consumer_description`: Brief description of the rollup BSN's purpose
+* `max_multi_staked_fps`: Maximum number of rollup FPs per BTC delegation. Prevents delegators from splitting
+ their BTC across too many networks (recommended: `3-5`)
+<!-- TODO: this should not be here, we will have a global limit on fps -->
+* `rollup_finality_contract_address`: Babylon Genesis address of the deployed Rollup BSN contract (`bbn1...` format)
+
+> **Note**: The above operation requires governance in permissioned
+> registration networks.
+
+## Contract Maintenance
+<img width="1848" height="677" alt="adminRole" src="https://github.com/user-attachments/assets/9b942068-3d72-4ae8-a47e-79894427204e" />
+
+The Rollup BSN contract exposes `admin-only` functions for ongoing management. These allow you to clean up
+ old data, update roles, and adjust configuration without redeploying the contract.
+
+Each operation is executed with:
+
+```shell
+babylond tx wasm execute <CONTRACT_ADDRESS> '<MSG>' \
+  --from $ADMIN_KEY \
+  --chain-id $BBN_CHAIN_ID \
+  --keyring-backend test \
+  --gas auto \
+  --gas-adjustment 1.3 \
+  --fees 1000000ubbn \
+  -y
+```
+
+### Modifying the Contract Administrator
+```shell
+MSG={
+  "update_admin": {
+    "admin": <NEW_ADMIN_ADDRESS>
+  }
+}
+```
+Transfers contract admin rights to a new Babylon address. Typically used when rotating keys or migrating
+ ownership to another entity
+
+### Data Pruning
+```shell
+MSG={
+  "prune_data": {
+    "rollup_height": <BLOCK_HEIGHT>,
+    "max_signatures_to_prune": <MAX_SIGS>, # optional
+    "max_pub_rand_values_to_prune": <MAX_PUB_RAND> # optional
+  }
+}
+```
+Removes all finality signatures and public randomness for rollup blocks with height less than or
+ equal to the specified value. Useful for reducing on-chain storage as the chain grows. Optional
+  fields control pruning batch size to avoid gas exhaustion.
```

Comment:
```suggestion
equal to the specified value. Useful for reducing on-chain storage as the chain grows. Optional
fields control pruning batch size to avoid gas exhaustion.
```